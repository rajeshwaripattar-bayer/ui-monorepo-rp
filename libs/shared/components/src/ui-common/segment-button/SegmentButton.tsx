import { useCallback, useState } from 'react'
import styles from './SegmentButton.module.scss'
import { Button, ButtonProps } from '@element/react-button'
import { Group } from '@element/react-group'

export interface SegmentButton {
  name: string
  value: string
}

export interface SegmentButtonProps {
  buttonProps?: ButtonProps & React.HTMLProps<HTMLElement>
  data: SegmentButton[]
  selectedValue?: string
  onClick?: (selectedValue: string) => void
}

export function SegmentButton(props: Readonly<SegmentButtonProps>) {
  const [active, setActive] = useState(props.selectedValue)

  const handleClick = useCallback(
    (selectedValue: string) => {
      setActive(selectedValue)
      props.onClick && props.onClick(selectedValue)
    },
    [props]
  )

  const buttons = props.data.map((button: SegmentButton) => (
    <Button
      {...props.buttonProps}
      label={button.name}
      key={button.name}
      variant={button.value === active ? 'filled' : 'outlined'}
      onClick={() => handleClick(button.value)}
    />
  ))

  return (
    <Group className={styles['container']} gap='none'>
      {buttons}
    </Group>
  )
}

export default SegmentButton
