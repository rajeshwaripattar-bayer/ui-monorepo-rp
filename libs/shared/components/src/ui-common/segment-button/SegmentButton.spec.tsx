import { render, fireEvent } from '@testing-library/react'
import SegmentButton from './SegmentButton'
import { noop } from 'lodash'

describe('SegmentButton', () => {
  const data = [
    { name: '$', value: '$' },
    { name: '%', value: '%' }
  ]
  it('should render successfully', () => {
    const { baseElement } = render(
      <SegmentButton buttonProps={{ buttonSize: 'xsmall' }} data={data} selectedValue='$' />
    )
    expect(baseElement).toBeTruthy()
  })

  it('Snapshot test', () => {
    const { baseElement } = render(
      <SegmentButton buttonProps={{ buttonSize: 'xsmall' }} data={data} selectedValue='$' />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('renders the correct name prop', () => {
    const { getByText } = render(
      <SegmentButton buttonProps={{ buttonSize: 'xsmall' }} data={data} selectedValue='$' onClick={noop} />
    )
    const dollar = getByText('$')
    expect(dollar).toBeTruthy()
    const percentage = getByText('%')
    expect(percentage).toBeTruthy()
  })

  it('handleClick', async () => {
    const handleClick = jest.fn()
    const { queryByText } = render(
      <SegmentButton buttonProps={{ buttonSize: 'xsmall' }} data={data} selectedValue='$' onClick={handleClick} />
    )

    const button = queryByText('%')
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('should render first button with border radius', () => {
    const { baseElement } = render(
      <SegmentButton buttonProps={{ buttonSize: 'xsmall' }} data={data} selectedValue='$' />
    )
    expect(baseElement.getElementsByClassName('first').length).toBe(0)
  })
  it('should render last button with border radius', () => {
    const { baseElement } = render(
      <SegmentButton buttonProps={{ buttonSize: 'xsmall' }} data={data} selectedValue='$' />
    )
    expect(baseElement.getElementsByClassName('last').length).toBe(0)
  })
})
