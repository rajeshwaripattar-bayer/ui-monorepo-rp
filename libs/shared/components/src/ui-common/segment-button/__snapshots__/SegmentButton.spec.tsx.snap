// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentButton Snapshot test 1`] = `
<body>
  <div>
    <div
      class="lmnt lmnt-group container"
      style="display: flex; flex-direction: row; justify-content: flex-start; align-items: baseline; gap: 0px;"
    >
      <button
        class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--filled-primary mdc-button--raised lmnt-button--xsmall mdc-touch-target-wrapper"
        type="button"
      >
        <span
          class="mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
          $
        </span>
        <span
          class="mdc-button__touch"
        />
      </button>
      <button
        class="lmnt mdc-button lmnt-button mdc-button--touch lmnt-button--outlined-primary mdc-button--outlined lmnt-button--xsmall mdc-touch-target-wrapper"
        type="button"
      >
        <span
          class="mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
          %
        </span>
        <span
          class="mdc-button__touch"
        />
      </button>
    </div>
  </div>
</body>
`;
